src/main/java/seedu/duke/tasks/Task.java

```diff
@@ -38,9 +46,16 @@ public void setTaskDone(boolean status) {
     @Override
     public String toString() {
         String taskStatusString = isTaskDone ? ICON_COMPLETED : ICON_UNCOMPLETED;
-        if (taskDescription == null) {
-            return String.format(TASK_STRING_NO_DESC, taskStatusString, taskName);
+        if (taskDescription != null && estimatedWorkingTime != null) {
+            return String.format(TASK_STRING_WITH_DESC_WITH_TIME, taskStatusString, taskName,
+                    taskDescription, estimatedWorkingTime);
+        } else if (taskDescription != null) {
+            return String.format(TASK_STRING_WITH_DESC_NO_TIME, taskStatusString, taskName, taskDescription);
+        } else if (estimatedWorkingTime != null) {
+            return String.format(TASK_STRING_NO_DESC_WITH_TIME, taskStatusString, taskName, estimatedWorkingTime);
+        } else {
+            return String.format(TASK_STRING_NO_DESC_NO_TIME, taskStatusString, taskName);
         }
```

Comment:
The save suggestion as that in the addParser part

Reply:
changed to switch case as suggested